src/controllers/waiterController.ts

```diff
@@ -0,0 +1,203 @@
+      Operation.Ok
+    );
+  } catch (e) {
+    await client.query("ROLLBACK");
+
+    logger("Failed to log in the waiter", e);
+
+    return sendResponse(res, "Something went wrong", Operation.ServerError);
+  } finally {
+    client.release();
+  }
+};
+
+export const waiterLogInConfirm = async (req: Request, res: Response) => {
+  const validatedReq = waiterLogInReq.safeParse(req.body);
+
+  if (!validatedReq.success) {
+    return sendResponse(res, "Invalid request", Operation.BadRequest);
+  }
+
+  const client = await pool.connect();
+  try {
+    await client.query("BEGIN");
+
+    const waiterData = await getFullWaiterData(validatedReq.data.email, client);
+
+    if (
+      waiterData &&
+      waiterData.minutesDifference >= confirmationPinValidityPeriod
+    ) {
+      await resetWaiterConfirmationDetails(validatedReq.data.email, client);
+      await client.query("COMMIT");
+
+      return sendResponse(res, "Expired pin code", Operation.Forbidden);
+    }
+
+    if (
+      !waiterData ||
+      !(await bcrypt.compare(validatedReq.data.pin, waiterData.confirmationPin))
+    ) {
+      await client.query("ROLLBACK");
+
+      return sendResponse(res, "Invalid credentials", Operation.BadRequest);
+    }
+
+    const { accessToken, refreshToken } = signTokens({
+      waiter_id: waiterData.id,
+      restaurant_id: waiterData.restaurantId,
+      waiter_email: waiterData.email,
+    });
+
+    await logInTheWaiter(validatedReq.data.email, refreshToken, client);
+
+    await client.query("COMMIT");
+
+    res.setHeader("Authorization", `Bearer ${accessToken}`);
+    return sendResponse(res, "Successful login", Operation.Ok);
+  } catch (e) {
+    await client.query("ROLLBACK");
+
+    logger("Failed to confirm waiter log in", e);
+
+    return sendResponse(
+      res,
+      "Failed to confirm your logging in",
+      Operation.ServerError
+    );
+  } finally {
+    client.release();
+  }
+};
+
+export const waiterLogOut = async (req: Request, res: Response) => {
+  try {
+    await deleteWaiterRefreshToken(req.waiter.waiter_id);
+
+    return sendResponse(res, "Successful log out", Operation.Ok);
+  } catch (e) {
+    logger("Failed to log out waiter", e);
+    return sendResponse(res, "Failed to log you out", Operation.ServerError);
+  }
+};
+
+export const renewSession = async (req: Request, res: Response) => {
+  const { accessToken, refreshToken } = signTokens({
+    waiter_id: req.waiter.waiter_id,
+    waiter_email: req.waiter.waiter_email,
+    restaurant_id: req.waiter.restaurant_id,
+  });
+
+  try {
+    await renewWaiterRefreshToken(refreshToken, req.waiter.waiter_id);
+
+    res.setHeader("Authorization", `Bearer ${accessToken}`);
+    return sendResponse(res, "Successfuly renewed your session", Operation.Ok);
+  } catch (e) {
+    logger("Failed to new waiter session", e);
+
+    return sendResponse(
+      res,
+      "Failed to renew your session",
+      Operation.ServerError
+    );
+  }
+};
```

Comment:
This controller should return user session. Maybe rename to getSession. On every request you can still renew, but also return user data